Rest/Controller/BaseController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\RecommendationBundle\Rest\Controller;
+
+use eZ\Publish\Core\REST\Server\Controller;
+use EzSystems\RecommendationBundle\Authentication\Authenticator;
+use EzSystems\RecommendationBundle\Rest\Content\ContentType;
+use Symfony\Component\HttpFoundation\Request;
+
+abstract class BaseController extends Controller
+{
+    /** @var \EzSystems\RecommendationBundle\Rest\Content\ContentType */
+    protected $contentType;
+
+    /** @var \EzSystems\RecommendationBundle\Authentication\Authenticator */
+    protected $authenticator;
+
+    /**
+     * @param \EzSystems\RecommendationBundle\Rest\Content\ContentType $contentType
+     * @param \EzSystems\RecommendationBundle\Authentication\Authenticator $authenticator
+     */
+    public function __construct(
+        ContentType $contentType,
+        Authenticator $authenticator
+    ) {
+        $this->contentType = $contentType;
+        $this->authenticator = $authenticator;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
```

Comment:
This would really deserve a lengthy description of what it does and what it is used for :)